### PR TITLE
Add manual extrinsic calibration

### PR DESCRIPTION
**Pull request by adthoms**
@nickcharron any questions let me know

This PR does the following:
* rename 3d_map_builder to map_builder to avoid build conflicts when creating a custom ROS message. This custom ROS message is required for the manual_calibration executable that is introduced in this PR
* refactor Load/Write interface for new updates introduced in the [modify_poses PR](https://github.com/BEAMRobotics/libbeam/pull/238) in libbeam
* add poses_to_poses_file executable and update documentation in README.md
* add manual_calibration executable and associated rviz and rqt_ez_publisher launch file. Update documentation in README.md

### CODE REVIEW

File: config/manual_calibration_config.yaml
Comment (nickcharron) [inline config/manual_calibration_config.yaml:4]: what does is_repeat do? can you add that to the docs?

File: README.md
Comment (nickcharron) [inline README.md:168]: save data just saves or publishes the cloud results? How do you save the calibration?

**Review by nickcharron — APPROVED** [review]
thanks for all this work! Most of it looks good at first glance, I haven't gone through it in detail but I'll stamp it anyways to unblock. I'd like to try out the calibrator myself soon though and I may have more questions or feedback then. Also, it would be cool to have a similar calibrator that works for cam-lidar using our colorizer code. I might need to do this in the near future

File: README.md
Comment (adthoms) [reply README.md:168]: I will change `save_data` to `publish` as this option simply publishes the clouds when any of the the parameters (i.e. `start_index`, `window_size`, `x,y,z,roll,pitch,yaw`) changes (see Algorithm 3 in the IEEE submission for specifics). Right now, there is no way of actually saving the calibration since this was sufficient for the paper. Ideally, there would be functionality to write calibrations to a file, though that would take some developement which I unfortunately cannot spend time on right now. The hand calculations take only a few minutes to do, though for someone not familiar with the code structure it would be a good feature.

File: config/manual_calibration_config.yaml
Comment (adthoms) [reply config/manual_calibration_config.yaml:4]: `is_repeat` essentially republishes the selected topic. This should always be set to false when using this tool as we only want to publish clouds when a parameter changes. I will make a note of this

**Comment by adthoms** [comment]
Before merging, should the current main branch be tagged a v1.0.0? Since these changes depend on a revised libbeam, we should finalize a v1.0.1 of libbeam and then have a corresponding  v1.0.1 tag for map_builder. Of course there is some refactoring required in the [Modify poses PR](https://github.com/BEAMRobotics/libbeam/pull/238) and ideally new tests would be made befiore a v1.0.1 tag of libbeam is made. A new docker container for libbeam would also have to be created, which should also be done in this PR before an official tag is made.

**Pull request merged**
